lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -80,38 +82,32 @@ impl RegionInternal for MmapRegion {
         module: Arc<dyn Module>,
         embed_ctx: CtxMap,
         heap_memory_size_limit: usize,
+        alloc_strategy: AllocStrategy,
     ) -> Result<InstanceHandle, Error> {
-        let custom_limits;
-        let mut limits = self.get_limits();
-
-        // Affirm that the module, if instantiated, would not violate
-        // any runtime memory limits.
-        match heap_memory_size_limit.cmp(&limits.heap_memory_size) {
-            Ordering::Less => {
-                // The supplied heap_memory_size is smaller than
-                // default. Augment the limits with this custom value
-                // so that it may be validated.
-                custom_limits = Limits {
-                    heap_memory_size: heap_memory_size_limit,
-                    ..*limits
-                };
-                limits = &custom_limits;
+        let limits = self.get_limits();
+        module.validate_runtime_spec(&limits, heap_memory_size_limit)?;
+
+        let slot;
+        match alloc_strategy {
+            AllocStrategy::Linear => {
+                slot = self
+                    .freelist
+                    .write()
+                    .unwrap()
+                    .pop()
+                    .ok_or(Error::RegionFull(self.capacity))?
             }
-            Ordering::Equal => (),
-            Ordering::Greater => {
-                return Err(Error::InvalidArgument(
-                    "heap memory size requested for instance is larger than slot allows",
-                ))
+            _ => {
```

Comment:
I think it would be preferable to use `AllocStrategy::Random` instead of `_`. That way when we add a new strategy, this gives you an informative error instead of defaulting to random.

Reply:
Agreed.  That is a bit of leftover cruft.  Thanks for catching this!